internal/field/internal/templates/element/mul_nocarry.go

```diff
@@ -1,57 +1,224 @@
 package element
 
-// MulNoCarry see https://hackmd.io/@gnark/modular_multiplication for more info on the algorithm
+// MulNoCarry see https://eprint.iacr.org/2022/1400.pdf annex for more info on the algorithm
+// Note that these templates are optimized for arm64 target, since x86 benefits from assembly impl.
 const MulNoCarry = `
 {{ define "mul_nocarry" }}
-var t [{{.all.NbWords}}]uint64
```

Comment:
Was there a performance issue with these? Were they heap allocated?

Reply:
correct -- with arm target, the array is allocated on the heap by the Go compiler.